parsing/pprintast.ml

```diff
@@ -1106,9 +1106,9 @@ and signature ctxt f x =  list ~sep:"@\n" (signature_item ctxt) f x
 and signature_item ctxt f x : unit =
   match x.psig_desc with
   | Psig_type (rf, l) ->
-      type_def_list ctxt f (rf, true, l)
+      type_def_list ctxt f (Some rf, true, l)
   | Psig_typesubst l ->
-      type_def_list ctxt f (Nonrecursive, false, l)
+      type_def_list ctxt f (None, false, l)
```

Comment:
There is no need to distinguish between `Recursive` and `None`, isn't it?
Switching to `Recursive` and adding a comment
```ocaml
(* Recursive is the default for type definition *)
type_def_list ctxt f (Recursive, false, l)
```
seems fine  enough to me?

Reply:
Fine by me